pages/app/course-catalog.js

```diff
@@ -1,10 +1,97 @@
 // ClassCatalog widget
+import { useEffect, useState } from 'react';
+import { Grid } from '@mui/material';
+import Stack from '@mui/material/Stack';
+import TextField from '@mui/material/TextField';
+import InputLabel from '@mui/material/InputLabel';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import Select from '@mui/material/Select';
+
+import CourseCatalogCard from '../../components/CourseCatalog/CourseCatalogCard.js'
 import MainButton from '../../components/basecomponents/MainButton.js';
+import MyCoursesExampleData from '../../components/MyCourses/data/MyCourses.example.js';
 
+// search based on: teacher name, class name
+// filter based on category or all
 export default function CourseCatalog() {
+  const [searchInput, setSearchInput] = useState('');
+  const [allCourses, setAllCourses] = useState(MyCoursesExampleData);
+  const [displayCourses, setDisplayCourses] = useState(allCourses);
+  const [currentCategory, setCurrentCategory] = useState('');
+  const [openCreateClass, setOpenCreateClass] = useState(false);
+
+  // after get request, set allCourses to results
+
+  const handleCreateClass = () => {
+    setOpenCreateClass(!openCreateClass);
+  };
+
+  const handleSearchInput = (e) => {
+    setSearchInput(e.target.value);
+  };
+
+  const handleChange = (event) => {
+    setCategory(event.target.value);
+  };
+
+  // if search input not empty, filter based on inputted course name/teacher name
+  const handleSearchSubmit = () => {
+    if (searchInput !== '') {
```

Comment:
Empty strings are falsy values, we can simply do if(searchInput), if your plan is to check if the field is empty or not